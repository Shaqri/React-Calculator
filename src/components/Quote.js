import React from 'react';
import Navbar from './Navbar';
import Text from './Text';
import * as styles from '../styles/components.module.css';

const Quote = () => {
  const quote = 'Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. â€“William Paul';
  return (
    <>
      <Navbar val="Quote" />
      <div className={`${styles.container} ${styles.quote}`}>
        <Text content={quote} />
      </div>
    </>
  );
};

export default Quote;
