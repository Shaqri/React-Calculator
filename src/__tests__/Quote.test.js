import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import React from 'react';
import Quote from '../components/Quote';
/* eslint-disable react/display-name */
jest.mock('../components/Navbar', () => () => <nav data-testid="navbar" />);

describe('Quote', () => {
  it('renders Navbar component', async () => {
    render(<Quote />);
    const navbar = await screen.getByTestId(/navbar/);
    expect(navbar).toBeInTheDocument();
    expect(navbar).toMatchSnapshot();
  });

  it('renders Text component with a quote', async () => {
    render(<Quote />);
    const text = await screen.getByText('Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. â€“William Paul');
    expect(text).toBeInTheDocument();
    expect(text).toMatchSnapshot();
  });
});
