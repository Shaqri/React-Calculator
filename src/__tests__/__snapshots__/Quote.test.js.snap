// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote renders Navbar component 1`] = `
<nav
  data-testid="navbar"
/>
`;

exports[`Quote renders Text component with a quote 1`] = `
<p>
  Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. â€“William Paul
</p>
`;
